sdk/core/core-rest-pipeline/documentation/pipeline.md

```diff
@@ -0,0 +1,220 @@
+- redirectPolicy
+  - A policy to follow Location headers from the server in order to support server-side redirection.
+- logPolicy
+  - A policy that logs all requests and responses.
+
+There are other policies commonly added to the default pipeline:
+
+- bearerTokenAuthenticationPolicy
+  - A policy that can request a token from a TokenCredential implementation and then apply it to the Authorization header of a request as a Bearer token.
+- serializationPolicy
+  - This policy handles assembling the request body and headers using an OperationSpec and OperationArguments on the request.
+- deserializationPolicy
+  - This policy handles parsing out responses according to OperationSpecs on the request.
+
+# Configuring client pipeline
+
+When instantiating a new client a pipeline can be configured by adding additional policies to the default pipeline, passing a static array of policies.
+
+```typescript
+const sendRequest = (request: PipelineRequest, next: SendRequest) => next(request);
+
+const policy1: PipelinePolicy = {
+  name: "policy1",
+  sendRequest,
+};
+const policy2: PipelinePolicy = {
+  name: "policy2",
+  sendRequest,
+};
+
+const client = new ServiceClient({
+  additionalPolicies: [
+    {
+      policy: policy1,
+      /**
+       * Determines if this policy be applied before or after retry logic.
+       * Only use `perRetry` if you need to modify the request again
+       * each time the operation is retried due to retryable service
+       * issues.
+       */
+      position: "perRetry",
+    },
+    { policy: policy2, position: "perCall" },
+  ],
+});
+```
+
+# Examples
+
+## Inserting a new policy at a specific place
+
+```typescript
+interface TracingPolicyOptions {
+  tracingEnabled: boolean;
+}
+
+function tracingPolicy(options: TracingPolicyOptions): PipelinePolicy {
+  return {
+    name: "TracingPolicy",
+    sendRequest: (request, next) => {
+      if (options.tracingEnabled) {
+        // add some headers
+        request.headers.set("tracing-id", "123456");
+      }
+
+      return next(request);
+    },
+  };
+}
+
+// Extend an existing Pipeline
+pipeline.addPolicy(tracingPolicy({ tracingEnabled: true }), { beforePhase: "Retry" });
+```
+
+## Removing policies
+
+Policies can be removed from a pipeline, consumers can target a specifc policy by name or remove all the policies in a specific phase
+
+```typescript
+// Remove a single policy, referencing it by name
+pipeline.removePolicy({ name: "customPolicy" });
+
+// Remove all policies from a phase
+pipeline.removePolicy({ phase: "Retry" });
+```
+
+## List policies
+
+Consumers can list all the policies in a pipeline calling `getOrderedPolicies()`, which returns the current set of policies in the pipeline in the order in which they will be applied to the request. Later in the list is closer to when the request is performed.
```

Comment:
Maybe mention this is really just for debugging purposes if you are confused by what order policies are being applied in. Again some remarks about bubbling could be good since "closer to when the request is performed" could be confused with when the request is initially made (the start of the pipeline) as opposed to when it is sent over the wire (the end of the pipeline).